generator/dart/packages/generated/google_cloud_rpc/test/json_test.dart

```diff
@@ -0,0 +1,147 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'dart:convert';
+
+import 'package:google_cloud_rpc/rpc.dart';
+import 'package:google_cloud_protobuf/protobuf.dart';
+import 'package:test/test.dart';
+
+// Test round-tripping generated JSON encoding and decoding code.
```

Comment:
I assume this file is hand-crafted and not generated (despite the directory).  We prefer to

- keep generated code separate from hand-crafted code: fewer temptations to "patch" the generated code, augment things "just for this case", and also easier to review and build tooling
- for tests like this, I would have a fully hand-crafted integration test that depends on the generated code and verifies it works